### PR TITLE
[codex] fix CI and CPU ST regressions

### PR DESCRIPTION
**Pull request by zhoubot**
## What changed
- fix the pre-commit failures introduced by the recent gitcode sync and normalize the affected markdown, Python, and C++ formatting
- restore CPU SIM full-ST runner behavior so testcase data generation and binary working directories match the single-test `run_cpu.py` path
- fix CPU SIM regressions exposed by the full suite, including `TFMOD` public wrapper behavior and `TINSERT` overload resolution
- align testcase golden-data generation for `trelu`, `tinsert`, and `texp` with the actual gtest case naming and lookup paths

## Why
The latest `main` branch failed the GitHub Actions `CI` workflow in two places:
- `pre-commit` failed on formatting and markdown lint issues in files brought over by sync work
- CPU SIM full ST had real regressions and runner mismatches, so local workflow-equivalent execution did not match the intended CI path

## Impact
- `pre-commit` is clean on the current diff
- docs build succeeds again
- CPU smoke and full CPU ST match the workflow behavior and pass locally

## Validation
- `pre-commit run --show-diff-on-failure --color=always --files $(git diff --name-only)`
- `cmake -S docs -B build/docs`
- `cmake --build build/docs --target pto_docs`
- `python3 tests/run_cpu.py --testcase tadd --gtest_filter 'TADDTest.case_float_64x64_64x64' --generator Ninja --build-dir build/cpu_smoke_tadd --clean`
- `bash tests/run_cpu_tests.sh --generator Ninja --build-folder build/cpu_st_ci`

Local result: CPU full ST `95/95` passed.

### CODE REVIEW

File: demos/auto_mode/baseline/add/test/test.py [hunk 1]
```diff
@@ -11,20 +11,16 @@
 # --------------------------------------------------------------------------------
 
 import torch
-import torch.nn as nn
-import torch_npu
 from torch_npu.testing.testcase import TestCase, run_tests
```
[the remark this answers is not in the document]
Comment (zhoubot) [reply demos/auto_mode/baseline/add/test/test.py:14]: Addressed in [PR #54](https://github.com/hw-native-sys/pto-isa/pull/54).

Fix details:
- restored `import op_extension` in `demos/auto_mode/baseline/add/test/test.py`
- marked it as an intentional side-effect import so lint cleanup does not remove it again

Root cause:
- `torch.ops.npu.my_add` is registered only when `op_extension/__init__.py` runs and loads `libop_extension.so`
- removing the import skipped that registration path, so the documented `python3 test.py` flow could fail with a missing `npu::my_add` operator before the test exercised the custom kernel

Validation:
- `python3 -m py_compile demos/auto_mode/baseline/add/test/test.py`
- `ruff check demos/auto_mode/baseline/add/test/test.py`

The PR is open and ready for review.

File: demos/auto_mode/torch_jit/add/add_compile_and_run.py [hunk 2]
```diff
@@ -11,7 +11,6 @@
 # --------------------------------------------------------------------------------
 
 import torch
-import torch_npu
 
 from jit_util_add import jit_compile
```
[the remark this answers is not in the document]
Comment (zhoubot) [reply demos/auto_mode/torch_jit/add/add_compile_and_run.py:15]: Addressed in [PR #55](https://github.com/hw-native-sys/pto-isa/pull/55).

Fix details:
- restored the explicit `import torch_npu` in `demos/auto_mode/torch_jit/add/add_compile_and_run.py`
- marked it as an intentional side-effect import so future cleanup does not remove it again

Root cause:
- this demo allocates tensors on `device="npu"` and uses `torch.npu.synchronize()`
- in standard `torch_npu` setups, importing `torch_npu` is what registers the `torch.npu` namespace and NPU dispatch integration
- removing that import could make the demo fail before kernel execution instead of testing the JIT-compiled add kernel

Validation:
- `python3 -m py_compile demos/auto_mode/torch_jit/add/add_compile_and_run.py`
- `ruff check demos/auto_mode/torch_jit/add/add_compile_and_run.py`

The PR is open and ready for review.

File: tests/cpu/st/testcase/trelu/gen_data.py [hunk 3]
```diff
@@ -64,33 +61,29 @@ def substring(a, b) -> str:
...
         case_params_list.append(TReluParams(BF16_DTYPE, 16, 256, 16, 256, 16, 256))
 
     for i, param in enumerate(case_params_list):
         case_name = generate_case_name(param)
-        if not os.path.exists(case_name):
-            os.makedirs(case_name)
+        if i < 8:
+            output_dir = os.path.join(script_dir, f"TRELUTest.case_{i}")
+        else:
+            output_dir = os.path.join(script_dir, case_name)
```
[the remark this answers is not in the document]
Comment (zhoubot) [reply tests/cpu/st/testcase/trelu/gen_data.py:84]: Addressed in [PR #56](https://github.com/hw-native-sys/pto-isa/pull/56).

Fix details:
- removed the index-based `i < 8` directory selection from `tests/cpu/st/testcase/trelu/gen_data.py`
- added explicit per-case output names to `TReluParams`
- routed the BF16 case through the same explicit mapping instead of relying on loop position

Why this fixes the issue:
- the previous implementation only worked while `case_params_list` stayed in the exact same order and size
- tying the output directory directly to the intended gtest case name makes the generated golden-data layout stable even if cases are inserted, removed, or reordered later

Validation:
- `python3 -m py_compile tests/cpu/st/testcase/trelu/gen_data.py`
- `ruff check tests/cpu/st/testcase/trelu/gen_data.py`
- `python3 tests/run_cpu.py --testcase trelu --gtest_filter 'TRELUTest.case_0' --generator Ninja --build-dir build/cpu_pr_trelu_dir_mapping --clean`

The PR is open and ready for review.

File: tests/script/all_cpu_tests.py [hunk 4]
```diff
@@ -124,27 +109,26 @@ def build_all_cpu_tests(repo_root: Path, build_dir: Path, args: argparse.Namespa
...
     gen_env = os.environ.copy()
     gen_env["PYTHONPATH"] = str(repo_root) + os.pathsep + gen_env.get("PYTHONPATH", "")
     if args.enable_bf16:
         gen_env["PTO_CPU_SIM_ENABLE_BF16"] = "1"
-    testcase_build_root.mkdir(parents=True, exist_ok=True)
     for script in sorted(testcase_src_root.glob("*/gen_data.py")):
-        run_command([sys.executable, str(script)], cwd=testcase_build_root, env=gen_env, verbose=args.verbose)
+        dst = build_dir / "gen_data.py"
+        shutil.copyfile(script, dst)
+        run_command([sys.executable, str(dst.name)], cwd=build_dir, env=gen_env, verbose=args.verbose)
```
[the remark this answers is not in the document]
Comment (zhoubot) [reply tests/script/all_cpu_tests.py:119]: Addressed in [PR #57](https://github.com/hw-native-sys/pto-isa/pull/57).

Fix details:
- stopped copying every testcase generator to the shared filename `build_dir/gen_data.py`
- now copy each generator to a unique filename derived from its testcase directory before executing it from the build directory

Why this fixes the issue:
- it removes the shared-filename collision point while preserving the existing build-directory execution model and generated data layout expected by the CPU-ST binaries
- this keeps the runner compatible with current tests while making it safer against future parallel execution or script assumptions around script identity

Validation:
- `python3 -m py_compile tests/script/all_cpu_tests.py`
- `ruff check tests/script/all_cpu_tests.py`
- `python3 tests/script/all_cpu_tests.py -g Ninja -b build/cpu_pr_all_cpu_tests`
  - result: `SUMMARY total=95 pass=95 fail=0`

The PR is open and ready for review.
